src/compat/libc/include/limits.h

```diff
@@ -118,4 +118,59 @@ Maximum number of bytes in a character, for any supported locale.
 //TODO life is pain
 #include <ctype.h>
 
+/* POSIX/SUS minimum requirements.
+ * These numbers come from SUS and are host independent. */
+
+#define _POSIX_AIO_LISTIO_MAX   2
+#define _POSIX_AIO_MAX          1
+#define _POSIX_ARG_MAX          4096
+#define _POSIX_CHILD_MAX        25
+#define _POSIX_CLOCKRES_MIN     20000000
+#define _POSIX_DELAYTIMER_MAX   32
+#define _POSIX_HOST_NAME_MAX    255
+#define _POSIX_LINK_MAX         8
+#define _POSIX_LOGIN_NAME_MAX   9
+#define _POSIX_MAX_CANON        255
+#define _POSIX_MAX_INPUT        255
+#define _POSIX_MQ_OPEN_MAX      8
+#define _POSIX_MQ_PRIO_MAX      32
+#define _POSIX_NAME_MAX         14
+#define _POSIX_NGROUPS_MAX      8
+#define _POSIX_OPEN_MAX         20
+#define _POSIX_PATH_MAX         256
+#define _POSIX_PIPE_BUF         512
+#define _POSIX_RE_DUP_MAX       255
+#define _POSIX_RTSIG_MAX        8
+#define _POSIX_SEM_NSEMS_MAX    256
+#define _POSIX_SEM_VALUE_MAX    32767
+#define _POSIX_SIGQUEUE_MAX     32
+#define _POSIX_SSIZE_MAX        32767
+#define _POSIX_STREAM_MAX       8
+#define _POSIX_SS_REPL_MAX      4
+#define _POSIX_SYMLINK_MAX      255
+#define _POSIX_SYMLOOP_MAX      8
+#define _POSIX_THREAD_DESTRUCTOR_ITERATIONS 4
+#define _POSIX_THREAD_KEYS_MAX  128
+#define _POSIX_THREAD_THREADS_MAX 64
+#define _POSIX_TIMER_MAX        32
+#define _POSIX_TRACE_EVENT_NAME_MAX 30
+#define _POSIX_TRACE_NAME_MAX   8
+#define _POSIX_TRACE_SYS_MAX    8
+#define _POSIX_TRACE_USER_EVENT_MAX 32
+#define _POSIX_TTY_NAME_MAX     9
+#define _POSIX_TZNAME_MAX       6
+#define _POSIX2_BC_BASE_MAX     99
+#define _POSIX2_BC_DIM_MAX      2048
+#define _POSIX2_BC_SCALE_MAX    99
+#define _POSIX2_BC_STRING_MAX   1000
+#define _POSIX2_CHARCLASS_NAME_MAX 14
+#define _POSIX2_COLL_WEIGHTS_MAX 2
+#define _POSIX2_EXPR_NEST_MAX   32
+#define _POSIX2_LINE_MAX        2048
+#define _POSIX2_RE_DUP_MAX      255
+
+#define _XOPEN_IOV_MAX          16
+#define _XOPEN_NAME_MAX         255
+#define _XOPEN_PATH_MAX         1024
```

Comment:
I am not sure about all these limits, since they look very complicated, and what if we cannot guarantee all of them satisfied?

For example, from https://pubs.opengroup.org/onlinepubs/7908799/xsh/limits.h.html
```
_POSIX_AIO_MAX
    The number of outstanding asynchronous I/O operations. Value: 1
```

How we can guarantee it would be so? In which way applications use these limits?

Reply:
Most of this are not actually used by the systems directly, they tend to be the minimum value that another constant can be. ie: _POSIX_PATH_MAX is the minimum value PATH_MAX can take. (in embox case, PATH_MAX is 256 as well, but in most systems it tends to be around 4096). 
The idea is to start laying the groundwork for future POSIX compliance. Obviously it will take time, but at the very least in the case of these defines, their inclusion really doesnt hurt. Of course, I could make an argument for each of them, however they're also a requirement for POSIX which doesnt really hurt to have at this stage.

Reply:
> The idea is to start laying the groundwork for future POSIX compliance. Obviously it will take time, but at the very least in the case of these defines, their inclusion really doesnt hurt. 

I agree to half of your statement. Ok, we can safely provide _POSIX_PATH_MAX. Nevertheless, I am not aware of the situations in which applications use `_POSIX_AIO_MAX` for example, like it was pointed above - https://github.com/embox/embox/pull/1775#discussion_r388808762 Let's imagine a situation when external application rely on the actual value of `_POSIX_AIO_MAX`, but we even will be not aware of that because it would be compiled well! In the case, if there are not such a macro in embox we will see compilation error, then we can examine why and how the app relies on this macro, then we will add and test this situation well.

Reply:
> > The idea is to start laying the groundwork for future POSIX compliance. Obviously it will take time, but at the very least in the case of these defines, their inclusion really doesnt hurt.
> 
> I agree with half of your statement. Well, we can safely provide _POSIX_PATH_MAX because we understand what does it mean and how we can guarantee the max path will be so. Nevertheless, I am not aware of the situations in which applications use `_POSIX_AIO_MAX` for example like it was pointed above - [#1775 (comment)](https://github.com/embox/embox/pull/1775#discussion_r388808762) Let's imagine a situation when external application relies on the actual value of `_POSIX_AIO_MAX`, but we even will be not aware of that because it would be compiled well! In case if there is no such a macro in embox we will see a compilation error, then we can examine why and how the app relies on this macro, then we will add and test this situation well.

I understand what you mean. Do you think that commenting out the limits currently not used and uncommenting them when needed might be a good compromise/middle ground solution, which 1) handles your valid concern regarding showing compilation errors during the porting effort, and 2) Still adds the entire list of minimum values for future reference?
If so, I could do that, and add a short comment stating that intention.

Reply:
Yes, you can either comment not used macros or just remove them and add a link to where we can get them if required.

Reply:
Ok, Ill so the former. Ive seen something like that done on unistd.h, I believe.